repos/system_upgrade/el7toel8/actors/satelliteupgrader/actor.py

```diff
@@ -0,0 +1,27 @@
+from leapp.actors import Actor
+from leapp.libraries.stdlib import api, run, CalledProcessError
+from leapp.models import SatelliteFacts
+from leapp.tags import FirstBootPhaseTag, IPUWorkflowTag
+
+
+class SatelliteUpgrader(Actor):
+    """
+    Execute installer in the freshly booted system, to finalize Satellite configuration
+    """
+
+    name = 'satellite_upgrader'
+    consumes = (SatelliteFacts, )
+    produces = ()
+    tags = (IPUWorkflowTag, FirstBootPhaseTag)
+
+    def process(self):
+        facts = next(self.consume(SatelliteFacts), None)
+        if not facts or not facts.has_foreman:
+            return
+
+        api.current_logger().info('Running the installer. This can take a while.')
+        try:
+            run(['foreman-installer'])
+        except CalledProcessError:
+            err_msg = 'Could not run the installer, please inspect the logs in /var/log/foreman-installer!'
+            api.current_logger().error(err_msg)
```

Comment:
```suggestion
        except OSError as e:
            api.current_logger().error('Failed to run `foreman-installer`: {}'.format(str(e))
        except CalledProcessError:
            api.current_logger().error(
                'Could not run the installer, please inspect the logs in /var/log/foreman-installer!'
            )
```

Reply:
In which case would this raise an `OSError` (just curious)?

Reply:
Problems with access rights (e.g. not +x), missing binary, anything before the process actually can be created